src/api/apiService.js

```diff
@@ -2,53 +2,73 @@ import { sessionService } from 'redux-react-native-session';
 import humps from 'humps';
 import Config from 'react-native-config';
 
-const saveSessionHeaders = (headers) => {
-  if (headers.get('access-token')) {
-    const sessionHeaders = {
-      token: headers.get('access-token'),
-      uid: headers.get('uid'),
-      client: headers.get('client')
-    };
-    sessionService.saveSession(sessionHeaders);
-  }
+import handleErrors from './utils/handleErrors';
+import getResponseBody from './utils/getResponseBody';
+
+const ACCESS_TOKEN = 'access-token';
+const APPLICATION_JSON = 'application/json';
+const CONTENT_TYPE = 'Content-Type';
+
+const HTTP_VERB = {
+  GET: 'get',
+  POST: 'post',
+  DELETE: 'delete',
+  PUT: 'put',
+  PATCH: 'patch',
 };
 
-const handleErrors = response =>
-  new Promise((resolve, reject) => {
-    if (!response) {
-      reject({ message: 'No response returned from fetch' });
-      return;
-    }
+class Api {
+  static get(uri, apiUrl = Config.API_URL) {
+    const requestData = this.buildRequestData(HTTP_VERB.GET);
+    return this.loadHeadersAndPerformRequest(uri, apiUrl, requestData);
```

Comment:
Never use `this` in static methods. `this` reference an instance.

```suggestion
    return Api.loadHeadersAndPerformRequest(uri, apiUrl, requestData);
```

Reply:
Actually calling `this` or `Api` in this case is the same. But i see that it could bring some confusion. I'll change it